### ISSUE TITLE
Block breaks in preview mode after changing the value of a field in the sidebar that affects WP_Query

### ISSUE THREAD

**Issue by joseph-farruggio** [AdvancedCustomFields/acf#734]
I launched a free course on ACF blocks and a student pointed out a bug in the block editor for a demo block we built.

The lesson they followed is here: https://www.modernwpdev.co/lessons/post-loop-block/

## The issue

We built a block that interfaces with WP_Query to display a list of posts.

While the block is in preview mode, you can change the "limit" for how many posts to display in the sidebar. If you change the limit once it works fine - you'll see the number of posts displayed change. 

But if you change the limit a second time the block breaks with the following error in the console:

``` html
react-dom.min.js?ver=17.0.1:9 TypeError: Cannot read properties of null (reading 'parentNode').
```

I've isolated this  issue to WP_Query. If I simply return the value of my limit field as text, the block's render template updates with no issues.

If I display the response from WP_Query, even just dumping the array with print_r(), the block breaks again.

## Steps to replicate
1. Create a block with a number field to set the "posts_per_page" and create a new instance of WP_Query in your render template
``` php
$queryArgs = array(
    'posts_per_page' => get_field( 'limit' ),
);

$posts = new WP_Query( $queryArgs );
```

2. Either create a loop for `$posts` or simply dump it out with `print_r($posts)`
3. Insert your block into the editor with Preview mode. Change the limit in the sidebar multiple times. The block should break.

**Comment by joseph-farruggio** [comment]
@lgladdy and I were chatting and he suspected what the issue was. My render template started with an HTML comment and this was what was breaking the block. What a strange bug, but it has nothing to do with the rest of my block's implementation.

**Comment by lgladdy** [comment]
This error can happen if the first thing inside your block template is a HTML comment, which seems to be the case here too!

This is because React creates this as a DOM element, but one it can't add a reference too - and we need that reference to figure out the parent.

We've got this logged as a bug though, and will try and get it fixed soon!

**Comment by CreativeDive** [comment]
That seems to be a related issue here. https://github.com/AdvancedCustomFields/acf/issues/721

**Comment by lgladdy** [comment]
This issue should be resolved in ACF 6.0.4 if the issue was caused by HTML comments in your template. If you continue to have issues, please let support know and we'll pick it up there!
